File: web-development/level-1/helsinki-part-1-2/Java/part1/week1-4/week01-04-ex-1-77/DecreasingCounter.java
public class DecreasingCounter  {
  //field
  private int value;
  private int initialNum;
  //constructor
  public DecreasingCounter(int initialNum)  {
    this.value = initialNum;
  }

  //behaviors
  public void printValue()  {
    System.out.println("This value: " + this.value);
  }

  public int decrease()  {
    if(this.value > 0)  {
      this.value--;
    }
    else if (this.value <= 0) {
      this.value = 0;
    }
    return this.value;
  }

  public void counterReset()  {
    this.value = 0;
  }

}

File: helsinki-moocs/python/week04/ex75/DecreasingCounter.py
class DecreasingCounter:

    #constructor
    def __init__(self, initial_value):
        self.value = initial_value
        self.initial_value = initial_value

    #behaviors
    def print_value(self):
        print('Value: ', self.value)

    def decrease(self):
        if self.value > 0:
            self.value -= 1
        else:
            print('cannot decrease below 0')

    def reset(self):
        self.value = 0

    def set_initial(self):
        self.value = self.initial_value
